### PR TITLE
Update tslib and typescript to 3.9

### PR DESCRIPTION
**Pull request by alan-agius4**

### CODE REVIEW

**Comment by alan-agius4** [comment]
Blocked on https://github.com/angular/angular-cli/pull/17760

File: modules/aspnetcore-engine/package.json [hunk 1]
```diff
@@ -16,14 +16,13 @@
     "ssr",
     "universal"
   ],
-  "engines": {
-    "node": ">=10.13.0"
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply modules/aspnetcore-engine/package.json:20]: Mistake 😱

File: modules/bazel-tsconfig-test.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./bazel-tsconfig-build.json",
   "compilerOptions": {
+    "target": "es5",
```
Comment (kyliau) [inline modules/bazel-tsconfig-test.json:4]: Could you please explain why this change?
Was `es5` already the default before?
or put it another way, could we use es2015?

**Comment by kyliau** [comment]
Let's cut rc.0 on Tuesday May 26 with this change :)

File: modules/bazel-tsconfig-test.json (hunk 2 above)
Comment (alan-agius4) [reply modules/bazel-tsconfig-test.json:4]: I actually thought I didn’t commit this.

Comment (alan-agius4) [reply modules/bazel-tsconfig-test.json:4]: Good catch @kyliau.
